File: javascript/leetcode/1252. Cells with Odd Values in a Matrix.js
/**
 * @param {number} n
 * @param {number} m
 * @param {number[][]} indices
 * @return {number}
 */
var oddCells = function (n, m, indices) {
  let odds = 0;

  const rows = [];
  const cols = [];
  rows.fill(n, false);
  cols.fill(m, false);

  indices.forEach((el) => {
    rows[el[0]] = !rows[el[0]];
    cols[el[1]] = !cols[el[1]];
  });

  for (let i = 0; i < n; ++i) {
    for (let j = 0; j < m; ++j) if (rows[i] ^ cols[j]) ++odds;
  }

  return odds;
};

File: c++/leetcode/1252. Cells with Odd Values in a Matrix.cpp
class Solution
{
public:
  int oddCells(int n, int m, vector<vector<int>> &indices)
  {
    int odds = 0;

    vector<bool> rows(n, false);
    vector<bool> cols(m, false);

    for (auto i : indices)
    {
      rows[i[0]] = !rows[i[0]];
      cols[i[1]] = !cols[i[1]];
    }

    for (int i = 0; i < n; ++i)
    {
      for (int j = 0; j < m; ++j)
      {
        if (rows[i] ^ cols[j])
          ++odds;
      }
    }

    return odds;
  }
};